### PR TITLE
Adding schemaRef support

### PR DESCRIPTION
**Pull request by sshefs02**

### CODE REVIEW

**Comment by nids** [comment]
@sshefs02  Please elaborate on the commit message

File: examples/src/main/java/com/adobe/platform/ecosystem/examples/catalog/model/SchemaRef.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        this.id = id;
+        this.contentType = contentType;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
```
Comment (vedhera) [inline examples/src/main/java/com/adobe/platform/ecosystem/examples/catalog/model/SchemaRef.java]: Return this from this method.

File: examples/src/main/java/com/adobe/platform/ecosystem/examples/constants/SDKConstants.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
     public static final String CONNECTION_IMS_ORG_KEY = "imsOrg";
     public static final String CONNECTION_HEADER_IMS_ORG_KEY = "x-gw-ims-org-id";
     public static final String CONNECTION_HEADER_JSON_CONTENT = "application/json";
+    public static final String CONNECTION_HEADER_JSON_FULL_CONTENT = "application/vnd.adobe.xed-full+json; version=1";
```
Comment (vedhera) [inline examples/src/main/java/com/adobe/platform/ecosystem/examples/constants/SDKConstants.java]: Remove this constant as we consume version from `schemaRef` present inside DataSet model

File: examples/src/test/java/com/adobe/platform/ecosystem/examples/catalog/model/DataSetTest.java [hunk 3]
```diff
@@ -326,6 +375,25 @@ public void testGetSchemaFieldsFromCatalog() throws ParseException, ConnectorSDK
...
+                anyString(),
+                anyString(),
+                anyObject(),
+                eq(false)
+        )).thenReturn(ds_Sample2.getFields(DataSet.FieldsFrom.OBSERVABLE_SCHEMA));
+
+        DataSet dataSet = new DataSetExtension(jsonObject);
+
+        final List<SchemaField> fieldList = dataSet.getFields(false, DataSet.FieldsFrom.SCHEMA);
+        //No blacklisting currently
```
Comment (vedhera) [inline examples/src/test/java/com/adobe/platform/ecosystem/examples/catalog/model/DataSetTest.java]: Remove comment.

**Pull request merged**